org.eclipse.january.test/src/org/eclipse/january/dataset/DatasetUtilsTest.java

```diff
@@ -0,0 +1,80 @@
+/*-
+ * Copyright (c) 2017 Kichwa Coders Ltd. and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.january.dataset;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class DatasetUtilsTest {
+
+	@Test
+	public void testRemoveNansAndInfinitiesDouble() {
+		double dvalue = 13;
+		Dataset dataset = DatasetFactory.createFromObject(
+				new Double[][] { { 10.0, 11.0, Double.NaN }, { Double.POSITIVE_INFINITY, Double.MAX_VALUE, 69.0 },
+						{ Double.MIN_VALUE, 0.0, Double.NEGATIVE_INFINITY } });
+		Dataset expected = DatasetFactory.createFromObject(new Double[][] { { 10.0, 11.0, dvalue },
+				{ dvalue, Double.MAX_VALUE, 69.0 }, { Double.MIN_VALUE, 0.0, dvalue } });
+		DatasetUtils.removeNansAndInfinities(dataset, dvalue);
+		assertEquals(dataset, expected);
```

Comment:
There are some tests utils for doing equals on datasets (TestUtils.assertDatasetEquals). Can you use that instead?